src/core/common/message.cpp

```diff
@@ -937,7 +957,7 @@ void PriorityQueue::Dequeue(Message &aMessage)
     aMessage.Next()         = nullptr;
     aMessage.Prev()         = nullptr;
 
-    aMessage.SetMessageQueue(nullptr);
+    aMessage.SetPriorityQueue(nullptr);
```

Comment:
This will set `mInPriorityQ = true`. Is this wanted behavior?

Reply:
Yes. When the queue is set to `nullptr` the `mInPriorityQ` flag does not matter.

We want to use `SetPriorityQueue(nullptr)` here since we are dequeuing the message from a `PriorityQueue`. The use of `SetMessageQueue()` here is not technically correct and relies on the knowledge of private/internal behavior of `Metdata` and how it re-using the same pointer for both queue type which `PriorityQueue` class should not really know.

Reply:
@simonlingoogle are you ok with this? Thanks.

Reply:
Ping @simonlingoogle .

Reply:
I am fine with this.